From e94108931a7ed75882b4bd5da463fee3f0f231f5 Mon Sep 17 00:00:00 2001
From: Daniel Caires <DanielMarques.Caires@windriver.com>
Date: Mon, 13 Jan 2025 10:40:59 -0300
Subject: [PATCH] Fix upversion breaking changes

After the OSH and OSH-I to caracal, two changes made into the upstream
broke the deployment of STX-O.

The first change it was a enconding of the password. This broke the
apply of STX-O because urlquery is a GO template function and when
the plugin tries to retrieve the password from the Helm environment
it is not encoded the same way.

The second change modified the pathtype of the Helm charts' ingress.

This patch reverts this two modification to the upstream.

Signed-off-by: Daniel Caires <DanielMarques.Caires@windriver.com>
---
 .../templates/endpoints/_authenticated_endpoint_uri_lookup.tpl  | 2 +-
 helm-toolkit/templates/manifests/_ingress.tpl                   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/helm-toolkit/templates/endpoints/_authenticated_endpoint_uri_lookup.tpl b/helm-toolkit/templates/endpoints/_authenticated_endpoint_uri_lookup.tpl
index d7390d8b..12b84dec 100644
--- a/helm-toolkit/templates/endpoints/_authenticated_endpoint_uri_lookup.tpl
+++ b/helm-toolkit/templates/endpoints/_authenticated_endpoint_uri_lookup.tpl
@@ -50,7 +50,7 @@ return: |
 {{- $endpointScheme := tuple $type $endpoint $port $context | include "helm-toolkit.endpoints.keystone_endpoint_scheme_lookup" }}
 {{- $userMap := index $context.Values.endpoints ( $type | replace "-" "_" ) "auth" $userclass }}
 {{- $endpointUser := index $userMap "username" }}
-{{- $endpointPass := index $userMap "password" | urlquery }}
+{{- $endpointPass := index $userMap "password" }}
 {{- $endpointHost := tuple $type $endpoint $context | include "helm-toolkit.endpoints.endpoint_host_lookup" }}
 {{- $endpointPort := tuple $type $endpoint $port $context | include "helm-toolkit.endpoints.endpoint_port_lookup" }}
 {{- $endpointPath := tuple $type $endpoint $port $context | include "helm-toolkit.endpoints.keystone_endpoint_path_lookup" }}
diff --git a/helm-toolkit/templates/manifests/_ingress.tpl b/helm-toolkit/templates/manifests/_ingress.tpl
index cacb4b81..4ea88966 100644
--- a/helm-toolkit/templates/manifests/_ingress.tpl
+++ b/helm-toolkit/templates/manifests/_ingress.tpl
@@ -625,7 +625,7 @@ examples:
 {{- $backendServiceType := index . "backendServiceType" -}}
 {{- $backendPort := index . "backendPort" -}}
 {{- $endpoint := index . "endpoint" | default "public" -}}
-{{- $pathType := index . "pathType" | default "Prefix" -}}
+{{- $pathType := index . "pathType" | default "ImplementationSpecific" -}}
 {{- $certIssuer := index . "certIssuer" | default "" -}}
 {{- $ingressName := tuple $backendServiceType $endpoint $envAll | include "helm-toolkit.endpoints.hostname_short_endpoint_lookup" }}
 {{- $backendName := tuple $backendServiceType "internal" $envAll | include "helm-toolkit.endpoints.hostname_short_endpoint_lookup" }}
-- 
2.34.1

